Vendor.Application/Requests/Markets/AddMarketCommand.cs

```diff
@@ -0,0 +1,47 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Vendor.Application.Requests.Vendor;
+using Vendor.Domain.Entities;
+
+namespace Vendor.Application.Requests.Markets
+{
+    public class AddMarketCommand: IRequest<Market>
+    {
+        public string Name { get; set; }
+    }
+    public class AddMarketCommandHandler : IRequestHandler<AddMarketCommand, Market>
+    {
+        private readonly VendorDbContext _dbContext;
+
+        public AddMarketCommandHandler(VendorDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+        public async Task<Market> Handle(AddMarketCommand request, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(request.Name))
```

Comment:
Can you do this validation using a separate validator file?

Reply:
done